src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
```

Comment:
I saw that in our code, we don't treat `isIframed` and `isInWebView` differently. If it's `isEmbedded`, then we have the messaging channel ready. I don't see why we want to treat them differently.